Supporting Resources/v2.0/Datamodel/Datafiles/Activity I-SD-3-B.yml

```diff
@@ -16,23 +16,24 @@ id:
 
 #The title of this activity
 title:
-  Files and repositories are checked periodically for secrets that should be protected.
+  Favor run-time exposure to secrets and improve lifecycle management.
 
 #Describe the benefit that is achieved by implementing this activity
 benefit: 
-  Risk of leaking production secrets is mitigated by removing all manual interactions and regular regeneration.
+  Risk of leaking production secrets is reduced by limiting access only during run-time and by regenerating regularly.
 
 #A one sentence description of the activity
 shortDescription:
-  Regenerate secrets dynamically during every deployment using well-hardened process wherever possible. 
+  Use run-time solutions to minimize access to application secrets. Properly manage the secret lifecycle.
 
 #A multi-paragraph description of the activity
 longDescription: 
-  Where secrets are not predefined or dependant on another system, generate them during the deployment process. Follow appropriate best practices such as using a cryptographically secure pseudorandom number generator if you generate this value randomly. Alert any manual access to secrets in the production environment. 
+  Whereever possible, rely on run-time techniques to manage access to the secrets required to execute the application. This is typically achieved by means of a vaulting solution with an quary interface that can be used by the application based on a bootstrapping secret.
```

Comment:
I cannot imagine anything particular under "rely on run-time techniques to manage access to the secrets" - do you have some links for this?

Reply:
Vaulting products have an API that the application container can query (based on some bootstrapping secret) to get a secret during run-time. That's what I'm referring to. The advantage is that the vault can actually manage the lifecycle at run-time (e.g., change a pwd on a DB server and give this new pwd to the requesting app).

Reply:
Ok, now I know what you're referring to, but I have a hard time to deduct it from the "run-time techniques" which are also referenced in the question, benefit etc. Maybe I'm the only one, could be discussed in the call.

Reply:
Reworded. I used the term on-demand, and the sentence after it explains it in more detail.